app/assets/javascripts/admin/products/services/variant_unit_manager.js.coffee

```diff
@@ -29,12 +29,17 @@ angular.module("admin.products").factory "VariantUnitManager", ->
           system: 'metric'
 
     @variantUnitOptions: ->
+      availableUnits = availableUnits.split(",")
       options = for unit_type, _ of @units
         for scale in @unitScales(unit_type)
           name = @getUnitName(scale, unit_type)
-          ["#{I18n.t(unit_type)} (#{name})", "#{unit_type}_#{scale}"]
+          if availableUnits.includes(name)
+            ["#{I18n.t(unit_type)} (#{name})", "#{unit_type}_#{scale}"]
+          else
+            null
```

Comment:
this method is now very complicated. I think we need to extract a part of it to a separate method to make it more simple.

Reply:
we would have a preferences api endpoint where we could expose quite a few of them including this one :tada:

Reply:
2a286af136c9a9365854afe5501123b37dce19ae